src/main/java/com/diffplug/gradle/osgi/BndManifestPlugin.java

```diff
@@ -97,36 +95,41 @@
 	protected void applyOnce(Project proj) {
 		ProjectPlugin.getPlugin(proj, JavaPlugin.class);
 		BndManifestExtension extension = proj.getExtensions().create(BndManifestExtension.NAME, BndManifestExtension.class);
-		Jar jarTask = (Jar) proj.getTasks().getByName(JavaPlugin.JAR_TASK_NAME);
-		// at the end of the jar, modify the manifest, and possibly write it to `osgiBndManifest { bndManifestcopyTo }`.
-		jarTask.doLast(unused -> {
-			Errors.rethrow().run(() -> {
-				byte[] manifest = getManifestContent(jarTask, extension).getBytes(StandardCharsets.UTF_8);
-				// modify the jar
-				Map<String, Function<byte[], byte[]>> toModify = ImmutableMap.of("META-INF/MANIFEST.MF", in -> manifest);
-				ZipMisc.modify(jarTask.getArchivePath(), toModify, Predicates.alwaysFalse());
-				// write manifest to the output resources directory
-				Throwing.Consumer<Path> writeManifest = path -> {
-					if (Files.exists(path)) {
-						if (Arrays.equals(Files.readAllBytes(path), manifest)) {
-							return;
+
+		proj.getGradle().getTaskGraph().whenReady(taskGraph -> {
+
+			// use all tasks which extends Jar
+			TaskCollection<Jar> jarTasks = proj.getTasks().withType(Jar.class);
+			// at the end of the jar, modify the manifest, and possibly write it to `osgiBndManifest { bndManifestcopyTo }`.
```

Comment:
I struggled a bit with this because if a task definition of e.g. ShadowJarPlugin is applied after BndManifestPlugin i will not get all the relevant tasks.

Reply:
Sorry, I misspoke.  Not `forEach`, but `all`.  `all` will be called for future tasks as well:

https://docs.gradle.org/current/javadoc/org/gradle/api/DomainObjectCollection.html#all-org.gradle.api.Action-